### PR TITLE
fix(content-sidebar): reply operations interactions

### PR DESCRIPTION
**Pull request by kkuliczkowski-box**
Changed the logic of hiding replies in a thread to removing them (leaving just the last one) from the Feed instead of displaying last one based on the internal `isExpanded` state in `ActivityThread`.

This allows to meet the requirement of displaying multiple replies when the thread is not expanded.

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js [hunk 1]
```diff
@@ -62,16 +64,44 @@ const ActivityThread = ({
     repliesTotalCount = 0,
     translations,
 }: Props) => {
-    const [isExpanded, setIsExpanded] = React.useState(isAlwaysExpanded);
+    const { length: repliesLength } = replies;
```
Comment (JChan106) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:67]: can replies length be 0?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js [hunk 2]
```diff
@@ -62,16 +64,44 @@ const ActivityThread = ({
     repliesTotalCount = 0,
```
Comment (JChan106) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:64]: What happens if this is defaulted to 0? Can we add tests for this edge case

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js [hunk 3]
```diff
@@ -62,16 +64,44 @@ const ActivityThread = ({
...
+                    className="bcs-ActivityThread-toggle"
+                    onClick={onShowReplies}
+                    type="button"
+                    data-testid="activity-thread-button"
+                >
+                    <FormattedMessage values={{ repliesToLoadCount }} {...messages.showReplies} />
+                </PlainButton>
+            );
+        }
+        if (repliesTotalCount > 1 && repliesTotalCount === repliesLength) {
```
Comment (JChan106) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:91]: So if replies total count is 1, we wont show any button? We should add a test for this too

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThreadReplies.js [hunk 4]
```diff
@@ -47,37 +43,21 @@ const ActivityThreadReplies = ({
 
     return (
         <div className="bcs-ActivityThreadReplies" data-testid="activity-thread-replies">
-            {!isExpanded ? (
+            {replies.map((reply: CommentType) => (
                 <Comment
-                    key={lastReply.type + lastReply.id}
-                    {...lastReply}
+                    key={reply.type + reply.id}
```
Comment (JChan106) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityThreadReplies.js:48]: can we use template literals for this? Seems more flexible and readable

File: src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityThread.test.js [hunk 5]
```diff
@@ -35,16 +39,28 @@ describe('src/elements/content-sidebar/activity-feed/activity-feed/ActivityThrea
...
+        const onHideReplies = jest.fn();
+        const lastReply = cloneDeep(repliesMock[repliesMock.length - 1]);
+        const { getByText } = getWrapper({ onHideReplies });
+
+        const button = getByText(localize(messages.hideReplies.id));
+        expect(button).toBeVisible();
+        fireEvent.click(button);
+
+        expect(onHideReplies).toBeCalledWith(lastReply);
     });
```
Comment (JChan106) [inline src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityThread.test.js:64]: Lets add some edge cases mentioned above where the total is 0 or the number of replies is 0 and vise versa

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js (hunk 1 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:67]: That is correct, when there are no replies the length is 0. Theoretically `onHideRepliesHandler` shouldn't be called when `repliesLength = 0,` but I've added additional check in `onHideRepliesHandler`.

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js (hunk 2 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:64]: It's 0 when there are no replies. In that case the button is not rendered. I've updated tests to cover that scenario.

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js (hunk 3 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js:93]: Correct. There was a test covering that, but I've made it now clearer to see.

File: src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityThread.test.js (hunk 5 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityThread.test.js:64]: I've modified the test for not rendering button with multiple cases. This test actually covers the rendering of button and calling `onHideReplies` on click. I've also updated the description so that it's clearer.

**Review by JChan106 — APPROVED** [review]
lgtm
